### PR TITLE
Make Request strict

### PR DESCRIPTION
**Pull request by rullzer**
Lets first see what CI thinks as I bet some stuff does :boom:

Todo:
- [ ] Some phpdocs lie saying it returns an array while it can also return null... fix?

### CODE REVIEW

**Comment by MorrisJobke** [comment]
> Some phpdocs lie saying it returns an array while it can also return null... fix?

I would keep it as it is now.



> Lets first see what CI thinks as I bet some stuff does 💥

Looks good 👍

**Review by MorrisJobke — APPROVED** [review]

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -175,8 +176,8 @@ public function setUrlParameters(array $parameters) {
 	 * Countable method
 	 * @return int
 	 */
-	public function count() {
-		return count(array_keys($this->items['parameters']));
+	public function count(): int {
+		return \count(array_keys($this->items['parameters']));
```
Comment (nickvergessen) [inline lib/private/AppFramework/Http/Request.php]: array_keys should not be needed?!

File: lib/private/AppFramework/Http/Request.php [hunk 2]
```diff
@@ -305,9 +311,9 @@ public function __unset($id) {
 	 * @param string $name
 	 * @return string
 	 */
-	public function getHeader($name) {
+	public function getHeader(string $name): string {
 
-		$name = strtoupper(str_replace(array('-'),array('_'),$name));
+		$name = strtoupper(str_replace(['-'], ['_'],$name));
```
Comment (nickvergessen) [inline lib/private/AppFramework/Http/Request.php]: can also remove the arrays

Comment (rullzer) [reply lib/private/AppFramework/Http/Request.php]: fixed

File: lib/private/AppFramework/Http/Request.php (hunk 1 above)
Comment (rullzer) [reply lib/private/AppFramework/Http/Request.php]: fixed

**Review by MorrisJobke — APPROVED** [review]

File: lib/private/AppFramework/Http/Request.php [hunk 3]
```diff
@@ -131,13 +132,13 @@ class Request implements \ArrayAccess, \Countable, IRequest {
 	 * @param string $stream
 	 * @see http://www.php.net/manual/en/reserved.variables.php
 	 */
-	public function __construct(array $vars=array(),
+	public function __construct(array $vars= [],
 								ISecureRandom $secureRandom = null,
 								IConfig $config,
 								CsrfTokenManager $csrfTokenManager = null,
-								$stream = 'php://input') {
+								string $stream = 'php://input') {
```
Comment (nickvergessen) [inline lib/private/AppFramework/Http/Request.php]: I love it how the parser still fucks up on this.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**